tests/test_ndarrays_regression.py

```diff
@@ -0,0 +1,599 @@
+            "    Max:     35.5",
+            "    Mean:    12.597121212121213",
+            "    Median:  3.55",
+            "  Individual errors:",
+            "          Index              Obtained              Expected            Difference",
+            "      (0, 0, 0)                    71                     0                    71",
+            "      (0, 0, 2)                    73                     2                    71",
+            "      (0, 0, 4)                    75                     4                    71",
+            "      (1, 0, 0)                    91                    20                    71",
+            "      (1, 0, 2)                    93                    22                    71",
+            "      (1, 0, 4)                    95                    24                    71",
+        ]
+    )
+    assert expected in obtained_error_msg
+
+
+def test_common_case_zero_expected(ndarrays_regression, no_regen):
+    # Most common case: Data is valid, is present and should pass
+    data = {"data1": np.array([0, 0, 2, 3, 0, 5, 0, 7])}
+    ndarrays_regression.check(data)
+
+    # Assertion error case: Only some zeros are not reproduced.
+    data = {"data1": np.array([1, 5, 2, 3, 0, 5, 3, 7])}
+    with pytest.raises(AssertionError) as excinfo:
+        ndarrays_regression.check(data)
+    obtained_error_msg = str(excinfo.value)
+    expected = "\n".join(
+        [
+            "Values are not sufficiently close.",
+            "To update values, use --force-regen option.",
+        ]
+    )
+    assert expected in obtained_error_msg
+    expected = "\n".join(
+        [
+            "data1:",
+            "  Shape: (8,)",
+            "  Number of differences: 3 / 8 (37.5%)",
+            "  Statistics are computed for differing elements only.",
+            "  Stats for abs(obtained - expected):",
+            "    Max:     5",
+            "    Mean:    3.0",
+            "    Median:  3.0",
+            "  Relative errors are not reported because all expected values are zero.",
+            "  Individual errors:",
+            "          Index              Obtained              Expected            Difference",
+            "              0                     1                     0                     1",
+            "              1                     5                     0                     5",
+            "              6                     3                     0                     3",
+        ]
+    )
+    assert expected in obtained_error_msg
+
+
+def test_different_data_types(ndarrays_regression, no_regen):
+    # Generate data with integer array.
+    data = {"data1": np.array([1] * 10)}
+    ndarrays_regression.check(data)
+
+    # Run check with incompatible type.
+    data = {"data1": np.array([True] * 10)}
+    with pytest.raises(
+        AssertionError,
+        match="Data types are not the same.\nkey: data1\nObtained: bool\nExpected: int64\n",
```

Comment:
Really minor, but, perhaps something like this would be a bit more readable?

```
match = "\n".join([
  "Data types are not the same.",
   "key: data1",
    "Obtained: bool"
    "Expected: int64"
])

Reply:
I agree.